### PR TITLE
Try prepare allocation without locking MERGEOK

### PR DESCRIPTION
**Pull request by freva**

### CODE REVIEW

**Review by mpolden — APPROVED** [review]
Great!

**Review by hmusum — APPROVED** [review]

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/GroupPreparer.java [hunk 1]
```diff
@@ -59,28 +61,26 @@ public GroupPreparer(NodeRepository nodeRepository,
...
-            try (Mutex allocationLock = nodeRepository.lockUnallocated()) {
+        // Try preparing in memory without lock. Most of the time there should be no changes and we can return nodes
+        // previously allocated.
+        {
+            MutableInteger probePrepareHighestIndex = new MutableInteger(highestIndex.get());
+            NodeAllocation probeAllocation = prepareAllocation(application, cluster, requestedNodes, surplusActiveNodes,
+                    probePrepareHighestIndex, wantedGroups, allocateFully, PROBE_LOCK);
+            if (probeAllocation.fulfilledAndNoChanges()) {
+                List<Node> acceptedNodes = probeAllocation.finalNodes();
+                surplusActiveNodes.removeAll(acceptedNodes);
```
Comment (hakonhall) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/GroupPreparer.java:73]: Why are nodes removed from surplusActiveNodes - I don't see any nodes are removed in the current/previous code?

Comment (mpolden) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/GroupPreparer.java:73]: The same thing is done in the existing code, at the end of the method.

Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/GroupPreparer.java:73]: Now I see it - never mind

Comment (freva) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/GroupPreparer.java:73]: Removed here:
https://github.com/vespa-engine/vespa/blob/90c8b5c969f16ba7ca0bc64f718a3ad8bdf527ff/node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/GroupPreparer.java#L119

Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/GroupPreparer.java:73]: Well, under the condition of dynamicProvisionEnabled

**Review by hakonhall — APPROVED** [review]

**Comment by hmusum** [comment]
Needed to fix critical deployment issues

**Pull request merged**